admission_control/admission-control-service/src/unit_tests/admission_control_service_test.rs

```diff
@@ -34,16 +34,15 @@ fn submit_transaction(
     ac_service: &MockAdmissionControlService,
     are_keys_valid: bool,
 ) -> SubmitTransactionResponse {
-    let keypair = compat::generate_keypair(None);
+    let mut rng = ::rand::rngs::StdRng::from_seed(TEST_SEED);
```

Comment:
This is `generate_for_testing`

Reply:
They generate secondary key if "are_keys_valid" is false. Actually if you look the test code was actually buggy so I wonder if it even tested anything... probably worth flagging to these folks, but I think this code has a limited shelf-life.